### PR TITLE
MDS-2842 Batch Downloading CRR PRR

### PR DESCRIPTION
**Pull request by areyeslo**
# Main

- What did you focus on in this PR?
Add Download All to download all files in one go in the following:
   - Code Required Reports
   - Permit Required Reports
   - Mine Act Permits

image

# Other
Existing Edit, Delete and Verify functionality are set together with Download All in dropdown under Actions button.

# How to test
- User is able to download all files when clicking on Download All under dropdown Actions.

# Notes
https://bcmines.atlassian.net/browse/MDS-2842

### CODE REVIEW

File: services/core-web/src/components/mine/Reports/MineReportTable.js [hunk 1]
```diff
@@ -173,6 +173,7 @@ export const MineReportTable = (props) => {
                 openEditReportModal={record.openEditReportModal}
                 handleEditReport={record.handleEditReport}
                 handleRemoveReport={record.handleRemoveReport}
+                minePermitRecord={record}
               />
```
Comment (mmumby) [inline services/core-web/src/components/mine/Reports/MineReportTable.js]: Can you rename this prop to be related to reports not permits? (`reportRecord={record}`)

Comment (areyeslo) [reply services/core-web/src/components/mine/Reports/MineReportTable.js]: Thanks a lot  for this requesting this change and others discussed in teams. This prop is removed. After the changes is no longer used.

File: services/core-web/src/components/mine/Permit/MinePermitTable.js [hunk 2]
```diff
@@ -80,7 +81,7 @@ const renderDeleteButtonForPermitAmendments = (record) => {
     return;
   }
 
-  const isLinkedToNowApplication = !isEmpty(record.amendmentEdit.amendment.now_application_guid);
+  const isLinkedToNowApplication = !isEmpty(record.operations.amendment.now_application_guid);
```
Comment (mmumby) [inline services/core-web/src/components/mine/Permit/MinePermitTable.js:85]: you should be able to read this as 'record.now_application_guid` as `amendment` has been spread into the object. 

Can you update and confirm?

File: services/core-web/src/components/mine/Permit/MinePermitTable.js [hunk 3]
```diff
@@ -406,38 +431,56 @@ const childColumns = [
...
-          </Button>
-        </AuthorizationWrapper>
-        {renderDeleteButtonForPermitAmendments(record)}
-      </div>
-    ),
+    render: (text, record) => {
+      const menu = (
+        <Menu>
+          <AuthorizationWrapper permission={Permission.ADMIN}>
+            <Menu.Item key="0">
```
Comment (mmumby) [inline services/core-web/src/components/mine/Permit/MinePermitTable.js:439]: this should be "EDIT_PERMITS"

File: services/core-web/src/components/mine/Permit/MinePermitTable.js [hunk 4]
```diff
@@ -500,14 +543,14 @@ const transformChildRowData = (
   permitAmendmentTypeOptionsHash
 ) => ({
   amendmentNumber,
-  amendmentType: amendment.permit_amendment_type_code,
+  // amendmentType: amendment.permit_amendment_type_code,
   isAmalgamated: amendment.permit_amendment_type_code === PERMIT_AMENDMENT_TYPES.amalgamated,
```
Comment (mmumby) [inline services/core-web/src/components/mine/Permit/MinePermitTable.js:542]: remove

File: services/core-web/src/components/mine/Reports/MineReportActions.js [hunk 5]
```diff
@@ -1,74 +1,143 @@
...
+                    this.props.mineReport
+                  )
+                }
+              >
+                <img
+                  src={EDIT_OUTLINE_VIOLET}
+                  alt="Edit Report"
+                  className="icon-sm padding-sm--right violet"
+                  style={{ paddingLeft: "13px" }}
+                />
```
Comment (mmumby) [inline services/core-web/src/components/mine/Reports/MineReportActions.js:116]: remove the additional padding, and use padding-md

File: services/core-web/src/components/mine/Permit/MinePermitTable.js (hunk 4 above)
Comment (areyeslo) [reply services/core-web/src/components/mine/Permit/MinePermitTable.js:542]: Removed.

File: services/core-web/src/components/mine/Permit/MinePermitTable.js (hunk 3 above)
Comment (areyeslo) [reply services/core-web/src/components/mine/Permit/MinePermitTable.js:439]: Changed.

File: services/core-web/src/components/mine/Permit/MinePermitTable.js (hunk 2 above)
Comment (areyeslo) [reply services/core-web/src/components/mine/Permit/MinePermitTable.js:85]: Updated.

**Review by mmumby — APPROVED** [review]
Nice work! Thanks for addressing all my comments.

TO PROD! 🔥

**Pull request merged**